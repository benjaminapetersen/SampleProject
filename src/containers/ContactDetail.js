/**
    @author Joseph Caiani

    @description Contact detail component displaying a single contact.

 */
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux'
import {Glyphicon, Grid, Col, Row} from 'react-bootstrap';
import {fetchContactDetail} from '../reducer/contacts/ContactReduce';
import {PageHeaderContent} from '../components/PageHeaderContent.js';
import {PanelContent} from '../components/PanelContent.js';

export class ContactDetail extends Component {
    constructor(props) {
        super(props);

        this.refresh = () => {
            const {dispatch, location} = this.props;
            dispatch(fetchContactDetail(parseInt(this.props.match.params.number.substr(1))));
        };
    }

    /**
     * componentDidMount
     * fetch the data when the component mounts
     * @param none
     */
    componentDidMount() {
        this.refresh();
    }

    /**
     * Define any props we want to use for this component
     * @returns {{}}
     */
    static get propTypes() {
        return {
            isContactDetailFetching: PropTypes.bool,
            dispatch: PropTypes.func.isRequired,
            contact: PropTypes.object,
            contactDetailError: PropTypes.any,
            location: PropTypes.any
        }
    }
    /**
     *   Render our component.
     *   @param none
     */
    render() {
        const {isContactDetailFetching, contact} = this.props;
        const contactData = isContactDetailFetching ? null : contact;
        return (
                <div>
                    {
                        contactData ?
                            <div>
                                <PageHeaderContent
                                    title={contactData ? contactData.name: ""}
                                    email={contactData ? contactData.email : ""}
                                />
                                <Grid className="detailGrid">
                                    <Row>
                                        <Col md={6}>
                                            <PanelContent
                                                title="Corporate Information"
                                                contact={contactData}
                                            />

                                        </Col>
                                        <Col md={6}>
                                            <PanelContent
                                                title="Personal Information"
                                                contact={contactData}
                                            />
                                        </Col>
                                    </Row>
                                </Grid>
                            </div>
                        :
                            <div className="configuringMessage">
                                <Glyphicon glyph="refresh"/> Loading...
                            </div>
                    }
                </div>

        );
    }
}

function mapStateToProps(state) {
    const {contactreduce} = state;

    const {
        contact: contact,
        isContactDetailFetching,
        contactsDetailError,
    } = contactreduce  || {
        contact: null,
        isContactDetailFetching: true,
        contactsDetailError: null
    };

    return {
        contact,
        isContactDetailFetching,
        contactsDetailError
    }

}
export default connect(mapStateToProps)(ContactDetail);
