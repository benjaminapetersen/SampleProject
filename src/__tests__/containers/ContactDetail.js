import React, {Component} from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {ContactDetail} from "../../containers/ContactDetail.js";
import { shallowToJson } from 'enzyme-to-json';
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });
// Make Enzyme functions available in all test files without importing
global.shallow = shallow;
global.render = render;
global.mount = mount;
// Fail tests on any warning
console.error = message => {
    throw new Error(message);
};

describe('ContactDetail', () => {
    it('should render correctly', () => {
        let mockDispatch = (() => {
            return null;
        });
        let mockMatch = {
            params: {
                number: ':3'
            }
        };
        const output = shallow(
            <ContactDetail dispatch={mockDispatch} match={mockMatch}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});