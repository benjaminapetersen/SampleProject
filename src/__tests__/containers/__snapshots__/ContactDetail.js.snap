// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactDetail should render correctly 1`] = `
<div>
  <div
    className="configuringMessage"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
     Loading...
  </div>
</div>
`;
